fundamentos/bloco-8-higher-order-functions-do-javascript-es6/dia-1-javascript-es6-introducao-a-higher-order-functions/bonus.js

```diff
@@ -0,0 +1,120 @@
+// Parte I - Game Actions Simulator
+
+const mage = {
+  name: 'Mago',
+  healthPoints: 130,
+  intelligence: 45,
+  mana: 75,
+  damage: undefined,
+};
+
+const warrior = {
+  name: 'Guerreiro',
+  healthPoints: 200,
+  strength: 30,
+  weaponDmg: 2,
+  damage: undefined,
+};
+
+const dragon = {
+  name: 'Dragão',
+  healthPoints: 350,
+  strength: 100,
+  damage: undefined,
+};
+
+const battleMembers = { mage, warrior, dragon };
+
+// 1 - Crie uma função que retorna o dano do dragão.
+// O dano será um número aleatório entre 15 (dano mínimo) e o valor do atributo strength (dano máximo).
+
+const dragonDamage = () => {
+  return Math.floor(Math.random() * (dragon['strength'] + 1 - 15 )) + 15
+}
+
+// 2 - Crie uma função que retorna o dano causado pelo warrior.
+// O dano será um número aleatório entre o valor do atributo strength (dano mínimo) e o valor de strength * weaponDmg (dano máximo).
+
+const warriorDamage = () => {
+  return Math.floor(Math.random() * (warrior['strength'] * warrior['weaponDmg'] + 1 - warrior['strength'])) + warrior['strength']
```

Comment:
Ainda vamos ver isso na Trybe, mas da uma olhada em desestruturação de objetos. ajuda bastante e deixa seu código mais limpo.
```suggestion
  const { strength, weaponDmg } = warrior;
  return Math.floor(Math.random() * (strength * weaponDmg + 1 - strength) + strength
```